### PR TITLE
Guillotine

### PR DESCRIPTION
**Pull request by Shadowmech88**
guillotine_open
Adds a guillotine.
Humans can be buckled into it. Buckling someone else into a guillotine takes 5 seconds. The human is able to unbuckle themselves unless the stocks are closed via a verb, in which case it'll take them 30 seconds to get out.
The rope can be untied in 10 seconds via Alt+click, which will drop the blade and decapitate the victim. Alt+clicking again will raise the blade.

WebM of it in action: http://puu.sh/u73de/aaf793a752.webm
Though it didn't at the time of recording, the guillotine now creates a blood splatter on its turf when someone is beheaded in it.

:cl:
 * rscadd: Added guillotines.

### CODE REVIEW

File: code/game/objects/structures/stool_bed_chair_nest/guillotine.dm [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+		to_chat(user, "<span class='warning'>You are too small to do that.</span>")
+		return
+
+	var/mob/M = get_locked(lock_type)[1]
+	if(M != user)
+		if(!open)
+			to_chat(user, "<span class='warning'>You can't pull \the [M] out of \the [src] while its stocks are closed.</span>")
+			return
+		else
+			M.visible_message(\
```
Comment (PJB3005) [inline code/game/objects/structures/stool_bed_chair_nest/guillotine.dm]: No backslashes here.

File: code/game/objects/structures/stool_bed_chair_nest/guillotine.dm [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+	if(iswrench(W))
+		if(anchored)
+			if(victim)
+				to_chat(user, "<span class='warning'>You can't unsecure \the [src] from the floor while someone's inside it!</span>")
+				return
+			to_chat(user, "You unsecure \the [src] from the floor.")
+		else
+			to_chat(user, "You secure \the [src] to the floor.")
+		playsound(get_turf(src), 'sound/items/Ratchet.ogg', 50, 1)
+		anchored = !anchored
```
Comment (PJB3005) [inline code/game/objects/structures/stool_bed_chair_nest/guillotine.dm]: The proc for a generic wrench based unanchor is obj level but I can't recall the name right now.

**Review by PJB3005 — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/stool_bed_chair_nest/guillotine.dm (hunk 2 above)
Comment (Kurfursten) [reply code/game/objects/structures/stool_bed_chair_nest/guillotine.dm]: wrenchanchor

**Comment by Ephx1** [comment]
Should give it a small chance to fail slicing cleanly, causing them to scream in agony and deal like 80 brute.

**Comment by CptWad** [comment]
dealing 80 brute to your head would decapitate it anyway

**Comment by Ephx1** [comment]
Oh well you get my point anyways, messing it up miserably, causing bleeding and screaming,

File: code/game/objects/structures/stool_bed_chair_nest/guillotine.dm (hunk 1 above)
Comment (Exxion) [reply code/game/objects/structures/stool_bed_chair_nest/guillotine.dm]: ?????
Who cares?

**Comment by FudgePucker** [comment]
IT GOES IT GOES IT GOES IT GOES

**Comment by Lokiamis** [comment]
Would it be possible to have the head just drop slightly instead of flying up a tile?

**Comment by calzilla1** [comment]
Only if you add a bucket

**Comment by eplgr** [comment]
I like the idea but I think it needs an animation and sounds.

edit: nevermind, after giving a look at the code I realized it does have sound (nice).
An animation would still be nice, though.

**Comment by Exxion** [comment]
Yo does this shit use resist yet

**Comment by diammz** [comment]
guillotine

https://my.mixtape.moe/xgrmve.dmi

**Comment by Blithering** [comment]
Can you make it so that you can leave a candle on the same tile, that will slowly burn through the rope? I want to twirl my moustache at the dastardly deathtrap while the mime plays dramatic music on a piano in the background

**Comment by Shadowmech88** [comment]
It's going to be tricky to get the blade animation to synch up with the decapitation, but I'll try.

**Comment by diammz** [comment]
im not much of a coder but you could split the animation into two parts, the blade falling and the cut+dangling rope and make it decapitate between the two

**Comment by PJB3005** [comment]
The issue is delays and timing with flick and such.

It's doable but very finnicky.

File: code/game/objects/structures/stool_bed_chair_nest/guillotine.dm (hunk 1 above)
Comment (PJB3005) [reply code/game/objects/structures/stool_bed_chair_nest/guillotine.dm]: I do.

It's ugly.

**Comment by Shadowmech88** [comment]
* Escaping from a guillotine is now handled by the resist verb.

**Comment by Shadowmech88** [comment]
* Animation added to guillotine.
WebM of new and improved guillotine, with sound: http://puu.sh/u73de/aaf793a752.webm

**Comment by diammz** [comment]
i see you improved the dangling rope animation, that looks much better

**Comment by Sprok0** [comment]
Does this have a delay before putting someone in it? If not, you can inescapably kill anyone nearby by putting them on it and closing the stocks via macro to skip the right-click menu.

**Comment by eplgr** [comment]
It now looks _really_ good.

**Comment by Shadowmech88** [comment]
I didn't think about macros.
I'll add a five-second delay to putting someone into the stocks.

**Comment by Shadowmech88** [comment]
* Placing someone into a guillotine now takes five seconds.

**Comment by ghost** [comment]
itgoesitgoesitgoesitgoesitgoesitgoesitgoesitgoes
/obscure reference

Looks pretty cool. Just needs a way to create it so it isn't adminbus only and then forgotten.

**Comment by Blithering** [comment]
Maybe wood block + an executioner's sword or the blade of one?

**Review by PJB3005 — APPROVED** [review]

**Pull request merged**